Add godoc-style docstrings for Cmd... methods

peggyl opened the pull request:
For #11596, checked using gofmt, golint, and godoc on localhost. 

This mostly just borrows the help messages from the CLI subcommand and flags parsers with light editing. Listing all the options in godoc felt rather heavy for subcommands with many flags, so this defers to the CLI help messages in the package description.

mfojtik committed:
Fix lxc-start in lxc>1.1.0 where containers start daemonized by default

Signed-off-by: Michal Fojtik <mfojtik@redhat.com>

aanm committed:
Dealing with trailing whitespaces

Created a validation that detects all trailing whitespaces from every
text file that isn't *.go, *.md, vendor/*,
docs/theme/mkdocs/tipuesearch*

Removed trailing whitespaces from every text file except from vendor/*
builder/parser/testfiles*, docs/theme/mkdocs/tipuesearch* and *.md

Signed-off-by: André Martins <martins@noironetworks.com>

Commit:
Moved pidfile from utils to pkg

Fixes #10958 by moving utils.daemon to pkg.pidfile.
Test cases were also added.
Updated the daemon to use the new pidfile.

Signed-off-by: Rick Wieman <git@rickw.nl>

brahmaroutu committed:
Fix the TestPullImageFromCentralRegistry to skip and add local v1 registry test when net=none

Closes #10966
Signed-off-by: Srini Brahmaroutu <srbrahma@us.ibm.com>

rhatdan committed:
Btrfs has eliminated the BTRFS_BUILD_VERSION in latest version

They say we should only use the BTRFS_LIB_VERSION

They will no longer support this since it had to be managed manually

Docker-DCO-1.1-Signed-off-by: Dan Walsh <dwalsh@redhat.com> (github: rhatdan)

jimmyxian committed:
fix docker ps help message

Signed-off-by: Xian Chaobo <xianchaobo@huawei.com>

rhatdan committed:
We want to allow the sharing of /dev from the host into the
 container.

docker run -v /dev:/dev should stop mounting other default mounts in i
libcontainer otherwise directories and devices like /dev/ptx get mishandled.

We want to be able to run libvirtd for launching vms and it needs
access to the hosts /dev.  This is a key componant of OpenStack.

Docker-DCO-1.1-Signed-off-by: Dan Walsh <dwalsh@redhat.com> (github: rhatdan)

hqhq committed:
fix decode data loss when using int64 in json

The problem is when I create container though REST api, set memory limit
in hostConfig, the memory limit didn't work. Because when we DecodeEnv,
we got hostConfig part of Env like this:
{"Binds":["/:/tmp"],"CpuShares":512,"CpusetCpus":"0,1","Devices":[],"Memory":1.6777216e+07,"MemorySwap":0}

And we cannot unmarshal number 1.6777216e+07 into Go value of type int64,
so we got 0.

We can fix this by setting Decoder as UseNumber().

Signed-off-by: Qiang Huang <h.huangqiang@huawei.com>

vbatts committed:
make.sh: leave around the generated version

For positerity (largely of packagers) lets leave around the generated
version files that happen during build.
They're already ignored in git, and recreated on every build.

Signed-off-by: Vincent Batts <vbatts@redhat.com>

Commit:
Return AuthResponse from postAuth api endpoint, Fixes #11607

Signed-off-by: Antonio Murdaca <me@runcom.ninja>

avidas committed:
Add comments to api/common constants, closes #11583

Signed-off-by: Avi Das <andas222@gmail.com>

Commit:
Removed unused "mutex" field

fixes #11659

Signed-off-by: Alena Prokharchyk <alena@rancher.com>

khooeee committed:
fix broken prerequisite link

Signed-off-by: Chris Khoo <chris.khoo@gmail.com>

afewremarks committed:
RE: Issue #6114.  Updated docs to reflect docker inpsect for volumes

Signed-off-by: Mark West <markewest@gmail.com>

calavera committed:
Use a structure to keep the allocated ips pool.

Fixes #11624.

Signed-off-by: David Calavera <david.calavera@gmail.com>

sdaingade committed:
Fixes error #11683

Signed-off-by: Swapnil Daingade <swapnil.daingade@gmail.com>

Commit:
Added documentation for specifying groupname or GID for commands. Also clarified used of the possible use of multiple USER commands in a Dockerfile.

Signed-off-by: Dan Anolik <dan@anolik.net>

jessedearing committed:
Add documentation about the semantics of `docker login`

Fixes #10550

Signed-off-by: Jesse Dearing <jesse.dearing@gmail.com>

khooeee committed:
fix test-and-docs typo "do" to "due"

Signed-off-by: Chris Khoo <chris.khoo@gmail.com>

Commit:
Add godoc-style docstrings for Cmd... methods

hqhq committed:
docs: add memory and swap memory usage examples

fix: https://github.com/docker/docker/issues/11629

Signed-off-by: Qiang Huang <h.huangqiang@huawei.com>

arthurbarr committed:
Fix #11589 by adding README and comments to exported functions

Signed-off-by: Arthur Barr <arthur.barr@uk.ibm.com>

paulbellamy committed:
Refactor global portallocator and portmapper state

Continuation of: #11660, working on issue #11626.

Wrapped portmapper global state into a struct. Now portallocator and
portmapper have no global state (except configuration, and a default
instance).

Unfortunately, removing the global default instances will break
```api/server/server.go:1539```, and ```daemon/daemon.go:832```, which
both call the global portallocator directly. Fixing that would be a much
bigger change, so for now, have postponed that.

Signed-off-by: Paul Bellamy <paul.a.bellamy@gmail.com>

hqhq committed:
add cpuset and examples to run.md

Signed-off-by: Qiang Huang <h.huangqiang@huawei.com>

Commit:
Refactor pkg/common, Fixes #11599

Signed-off-by: Antonio Murdaca <me@runcom.ninja>

4gh committed:
add tests for pkg/httputils

closes #11597

Signed-off-by: Frank Herrmann <fgh@4gh.tv>

Commit:
Sort .gitignore content

Signed-off-by: Arnaud Porterie <arnaud.porterie@docker.com>

HuKeping committed:
Fix minor typo

Fix minor typo and make the comments of version-comparison functions
uniform.

Signed-off-by: Hu Keping <hukeping@huawei.com>

Commit:
Add some info about what environment variables are available

Having the list in one spot makes it easier for people to see what's
avaiable instead of having to scan all of the docs and extract the info.

Signed-off-by: Doug Davis <dug@us.ibm.com>

LK4D4 committed:
Use /var/run/docker as root for execdriver

Signed-off-by: Alexander Morozov <lk4d4@docker.com>

Commit:
Adding in comments from party

Signed-off-by: Mary Anthony <mary@docker.com>

noxiouz committed:
History.Swap Use parallel assignment to swap elements, as it's

a more idiomatic way for golang than using a temp variable.

Signed-off-by: Anton Tiurin <noxiouz@yandex.ru>

Commit:
Fix volume initialize error check, Fixes #11725

Signed-off-by: Antonio Murdaca <me@runcom.ninja>

crosbymichael committed:
Mkdir for lxc root dir before setup of symlink

Signed-off-by: Michael Crosby <crosbymichael@gmail.com>

Commit:
Revert all but TestPullImageFromCentralRegistry changes

Signed-off-by: Arnaud Porterie <arnaud.porterie@docker.com>

jessfraz committed:
Get rid of panic in stats for lxc

Fix containers dir

Docker-DCO-1.1-Signed-off-by: Jessie Frazelle <jess@docker.com> (github: jfrazelle)

sirupsen committed:
listenbuffer: add docs

Signed-off-by: Simon Eskildsen <sirup@sirupsen.com>

sirupsen committed:
listenbuffer: add test

Signed-off-by: Simon Eskildsen <sirup@sirupsen.com>

Commit:
TestBuildCancelationKillsSleep send exec cmd to stdout
and makes the testing output ugly.

This hides the output since it not used.

Signed-off-by: Doug Davis <dug@us.ibm.com>

jessfraz committed:
fix 2 integration tests on lxc

Docker-DCO-1.1-Signed-off-by: Jessica Frazelle <jess@docker.com> (github: jfrazelle)

aneshas committed:
Fixed redundant else in GetDeviceStatus

Signed-off-by: Anes Hasicic <anes.hasicic@gmail.com>

GordonTheTurtle commented:
Can you please sign your commits following these rules:

https://github.com/docker/docker/blob/master/CONTRIBUTING.md#sign-your-work

The easiest way to do this is to amend the last commit:

~~~console
$ git clone -b "11596-api-commands-godoc" git@github.com:peggyl/docker.git somewhere
$ cd somewhere
$ git commit --amend -s --no-edit
$ git push -f
~~~
This will update the existing PR, so you do not need to open a new one.

GordonTheTurtle commented:
Looks like we would not be able to merge this PR because of merge conflicts. Please fix them and force push to your branch.

peggyl commented:
Sorry, my bad – forgot to rebase just now and pull in 58690c9cca5995035b721ed6f1337b0ddb932d7a especially. If docstrings look fine, I'll port them to the individual command files.

Commit:
Fix login

Right now it returns:
```
FATA[0001] json: Unmarshal(non-pointer types.AuthResponse)
```

Signed-off-by: Doug Davis <dug@us.ibm.com>

peggyl committed:
Add godoc-style docstrings to Cmd... methods

Signed-off-by: Peggy Li <peggyli.224@gmail.com>

peggyl committed:
Delete commands.go

peggyl committed:
Rebase
